test/e2e/scenario/recorder.scenario.ts

```diff
@@ -79,25 +81,512 @@ describe('recorder', () => {
 
         const fullSnapshot = findFullSnapshot(events.sessionReplay[0].segment.data)!
 
-        const fooNode = findNodeWithId(fullSnapshot, 'foo')
+        const fooNode = findElementWithIdAttribute(fullSnapshot, 'foo')
         expect(fooNode).toBeTruthy()
         expect(findTextContent(fooNode!)).toBe('foo')
 
-        const barNode = findNodeWithId(fullSnapshot, 'bar')
+        const barNode = findElementWithIdAttribute(fullSnapshot, 'bar')
         expect(barNode).toBeTruthy()
         expect(barNode!.attributes['data-dd-privacy']).toBe('hidden')
         expect(barNode!.childNodes.length).toBe(0)
 
-        const bazNode = findNodeWithId(fullSnapshot, 'baz')
+        const bazNode = findElementWithIdAttribute(fullSnapshot, 'baz')
         expect(bazNode).toBeTruthy()
         expect(bazNode!.attributes.class).toBe('dd-privacy-hidden baz')
         expect(bazNode!.attributes['data-dd-privacy']).toBe('hidden')
         expect(bazNode!.childNodes.length).toBe(0)
       })
   })
 
+  describe('mutations observer', () => {
+    createTest('record mutations')
+      .withSetup(bundleSetup)
+      .withRumRecorder()
+      .withBody(
+        html`
+          <p>mutation observer</p>
+          <ul>
+            <li></li>
+          </ul>
+        `
+      )
+      .run(async ({ events }) => {
+        await browserExecute(() => {
+          const li = document.createElement('li')
+          const ul = document.querySelector('ul') as HTMLUListElement
+          ul.appendChild(li)
+          document.body.removeChild(ul)
+          const p = document.querySelector('p') as HTMLParagraphElement
+          p.appendChild(document.createElement('span'))
```

Comment:
FMU, you are adding the `li` to be sure that there is no mutations captured due to the parent removal.
If that is it, it could be nice to be more explicit about it.